webroot/src/components/Licensee/LicenseeSearch/LicenseeSearch.ts

```diff
@@ -0,0 +1,180 @@
+//
+//  LicenseeSearch.ts
+//  CompactConnect
+//
+//  Created by InspiringApps on 9/12/2024.
+//
+
+import { Component, mixins, toNative } from 'vue-facing-decorator';
+import { reactive, computed } from 'vue';
+import MixinForm from '@components/Forms/_mixins/form.mixin';
+import InputText from '@components/Forms/InputText/InputText.vue';
+import InputSelect from '@components/Forms/InputSelect/InputSelect.vue';
+import InputSubmit from '@components/Forms/InputSubmit/InputSubmit.vue';
+import SearchIcon from '@components/Icons/Search/Search.vue';
+import { CompactType } from '@models/Compact/Compact.model';
+import { FormInput } from '@models/FormInput/FormInput.model';
+import Joi from 'joi';
+
+export interface LicenseSearch {
+    firstName?: string;
+    lastName?: string;
+    ssn?: string,
+    state?: string,
+}
+
+@Component({
+    name: 'LicenseeSearch',
+    components: {
+        InputText,
+        InputSelect,
+        InputSubmit,
+        SearchIcon,
+    },
+    emits: [ 'searchParams' ],
+})
+class LicenseeSearch extends mixins(MixinForm) {
+    //
+    // Lifecycle
+    //
+    created() {
+        this.initFormInputs();
+    }
+
+    //
+    // Computed
+    //
+    get userStore() {
+        return this.$store.state.user;
+    }
+
+    get compactType(): CompactType | null {
+        return this.userStore.currentCompact?.type;
+    }
+
+    get stateOptions(): Array<any> {
+        const { currentCompact } = this.userStore;
+        const compactMemberStates = (currentCompact?.memberStates || []).map((state) => ({
+            value: state.abbrev, name: state.name()
+        }));
+        const defaultSelectOption: any = { value: '' };
+
+        if (!compactMemberStates.length) {
+            defaultSelectOption.name = '';
+        } else {
+            defaultSelectOption.name = computed(() => this.$t('common.selectOption'));
+        }
+
+        compactMemberStates.unshift(defaultSelectOption);
+
+        return compactMemberStates;
+    }
+
+    //
+    // Methods
+    //
+    initFormInputs(): void {
+        this.formData = reactive({
+            firstName: new FormInput({
+                id: 'first-name',
+                name: 'first-name',
+                label: computed(() => this.$t('common.firstName')),
+                placeholder: computed(() => this.$t('licensing.searchPlaceholderName')),
+                validation: Joi.string().min(0).max(10).messages(this.joiMessages.string),
```

Comment:
Just curious: Is there a reason we don't use zod?

Reply:
Our team encounters plenty of cases where we're not using typescript. As a result, we're much more familiar with Joi.